core/unix/os.c

```diff
@@ -1291,12 +1297,21 @@ query_time_seconds(void)
     }
 }
 
+#if defined(MACOS) && defined(AARCH64)
+#    include <sys/time.h>
+#endif
+
 /* milliseconds since 1601 */
 uint64
 query_time_millis()
 {
     struct timeval current_time;
+#if !(defined(MACOS) && defined(AARCH64))
     uint64 val = dynamorio_syscall(SYS_gettimeofday, 2, &current_time, NULL);
+#else
+    uint64 val = gettimeofday(&current_time, NULL);
```

Comment:
As above, library routines cause problems.  Please include a XXX i#NNN comment about this being temporary.